src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object shape) {
+        if (shape == this) {
+            return true;
+        }
+        if (shape == null) {
+            return false;
+        }
+        if (shape.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) shape;
+            return ((this.length == current.length
+                    || this.length != null && this.length.equals(current.length)))
+                    && ((this.width == current.width
+                    || this.width != null && this.width.equals(current.width)))
+                    && ((this.color == current.color
+                    || this.color != null && this.color.equals(current.color)));
```

Comment:
remove this keyword
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1607422728299900

Reply:
resolved!